lib/ansible/modules/remote_management/foreman/foreman_global_parameter.py

```diff
@@ -0,0 +1,198 @@
+      name: "TheAnswer"
+      value: "42"
+      state: present
+
+- name: "Delete a Global Parameter"
+  local_action:
+      module: foreman_global_parameter
+      username: "admin"
+      password: "admin"
+      server_url: "https://fakeserver.com"
+      name: "TheAnswer"
+      state: absent
+'''
+
+RETURN = '''# '''
+
+try:
+    from nailgun.config import ServerConfig
+    import nailgun.entity_mixins
+    import nailgun.entities
+    HAS_NAILGUN_PACKAGE = True
+except:
+    HAS_NAILGUN_PACKAGE = False
+
+from ansible.module_utils.basic import AnsibleModule
+
+'''CommonParameter needs to inherit functionalities
+   for searching, updating and deleting'''
+
+
+class CommonParameter(
+    nailgun.entities.CommonParameter,
+    nailgun.entity_mixins.Entity,
+    nailgun.entity_mixins.EntityCreateMixin,
+    nailgun.entity_mixins.EntityDeleteMixin,
+    nailgun.entity_mixins.EntityReadMixin,
+    nailgun.entity_mixins.EntitySearchMixin,
+    nailgun.entity_mixins.EntityUpdateMixin
+):
+
+    pass
+
+
+def create_CommonParameter(name, value):
+    CommonParameter(name=name, value=value).create()
+
+
+def update_CommonParameter(name, value):
+    CommonParameter(name=name, value=value).update()
+
+
+def create_Server(server_url, auth, verify_ssl):
+    ServerConfig(
+        url=server_url,
+        auth=auth,
+        verify=verify_ssl,
+    ).save()
+
+
+def find_Entity(entity, **kwargs):
+    instance = entity(**kwargs)
+    return instance.search(
+        set(),
+        {'search': '%s="%s"' % (key, kwargs[key]) for key in kwargs}
+    )
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            server_url=dict(required=True),
+            username=dict(required=True, no_log=True),
+            password=dict(required=True, no_log=True),
+            verify_ssl=dict(required=False, type='bool', default=False),
+            name=dict(required=True, no_log=False),
+            value=dict(required=False, no_log=False),
+            state=dict(required=True, no_log=False),
+        ),
+        supports_check_mode=True
+    )
+    if not HAS_NAILGUN_PACKAGE:
+        module.fail_json(
+            msg="""Missing required nailgun module (check docs or install
+            with: pip install nailgun"""
+        )
+
+    server_url = module.params['server_url']
+    username = module.params['username']
+    password = module.params['password']
+    verify_ssl = module.params['verify_ssl']
+    name = module.params['name']
+    value = module.params['value']
+    state = module.params['state']
+    auth = (username, password)
+    changed = False
+    try:
+        create_Server(server_url, auth, verify_ssl)
+        param = find_Entity(CommonParameter, name=name)
+    except Exception as e:
+        module.fail_json(msg="Failed to connect to Foreman server: %s " % e)
+    try:
+        if state == 'present':
+            if not value:
+                module.fail_json(msg='value is required when state=present')
```

Comment:
There are built in helpers for this,

See the following draft example https://github.com/gundalow/ansible/blob/docs-argspec/docs/docsite/rst/dev_guide/developing_modules_general.rst#main-and-ansiblemodule-argument-spec